app/src/main/java/org/wikipedia/main/MainActivity.kt

```diff
@@ -61,10 +62,20 @@ class MainActivity : SingleFragmentActivity<MainFragment>(), MainFragment.Callba
     }
 
     override fun onResume() {
+        maybeShowPlacesSurvey()
         super.onResume()
         invalidateOptionsMenu()
     }
 
+    private fun maybeShowPlacesSurvey() {
+        binding.root.postDelayed({
+        if (Prefs.shouldShowOneTimePlacesSurvey == PlacesFragment.SHOW_SURVEY) {
+            Prefs.shouldShowOneTimePlacesSurvey = PlacesFragment.DO_NOT_SHOW_SURVEY
```

Comment:
When calling `postDelayed` we always want to make sure the activity isn't destroyed. If the activity is exited within 1 second, this will crash (silently), fail to show the survey, and put the survey into an incorrect state.